CMakeLists.txt

```diff
@@ -3,7 +3,7 @@ project(qxmpp VERSION 1.4.0)
 
 set(SO_VERSION 3)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
I'd probably make this a requirement:
```cmake
set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
```

While we're on this, what about disabling the language extensions (like GNU ones on gcc/clang) to the C++ standard for the sake of portability?
```cmake
set(CMAKE_CXX_EXTENSIONS OFF)
```

Reply:
Yes, both points seem to be a good idea. :)